src/test/java/com/app/onlinebookstore/controller/BookControllerTest.java

```diff
@@ -0,0 +1,199 @@
+package com.app.onlinebookstore.controller;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.security.test.web.servlet.setup.SecurityMockMvcConfigurers.springSecurity;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import com.app.onlinebookstore.dto.book.BookDto;
+import com.app.onlinebookstore.dto.book.CreateBookRequestDto;
+import com.app.onlinebookstore.model.Category;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Set;
+import org.apache.commons.lang3.builder.EqualsBuilder;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.test.context.support.WithMockUser;
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@Sql(scripts = "classpath:database/create/add-book-and-category.sql",
+        executionPhase = Sql.ExecutionPhase.BEFORE_TEST_METHOD)
+@Sql(scripts = "classpath:database/delete/delete-book-and-category-data.sql",
+        executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+public class BookControllerTest {
+    protected static MockMvc mockMvc;
+    private static final CreateBookRequestDto CREATE_BOOK_REQUEST_DTO = new CreateBookRequestDto();
+    private static final BookDto BOOK_DTO = new BookDto();
+    private static final Category CATEGORY = new Category();
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @BeforeAll
+    static void beforeAll(
+            @Autowired WebApplicationContext applicationContext) {
+        mockMvc = MockMvcBuilders
+                .webAppContextSetup(applicationContext)
+                .apply(springSecurity())
+                .build();
+
+        CATEGORY.setId(1L);
+        CATEGORY.setName("Test Category 1");
+        CATEGORY.setDescription("Description for Test Category 1");
+
+        CREATE_BOOK_REQUEST_DTO.setTitle("Test Title1");
```

Comment:
Please create constants wich fields will be populated and reuse them in test methods